test/suites/osconfig/clusterid.robot

```diff
@@ -83,10 +102,79 @@ Get MicroShift Cluster ID From Namespace
+    ...    sudo=True    return_rc=True    return_stdout=True
+    Log    ${sos_report_dir}
+    Should Be Equal As Integers    0    ${rc}
+
+    ${rc}=    Execute Command
+    ...    tar xf ${sos_report_tarfile} -C ${sos_report_dir}
+    ...    sudo=True    return_rc=True    return_stdout=False
+    Should Be Equal As Integers    0    ${rc}
+
+    ${sos_report_untared}    ${rc}=    Execute Command
+    ...    find ${sos_report_dir} -type d -name "sosreport-*"
+    ...    sudo=True    return_rc=True    return_stdout=True
+    Should Be Equal As Integers    0    ${rc}
+
+    RETURN    ${sos_report_untared}
+
+Get MicroShift Cluster ID From Sos Report
+    [Documentation]    Read and return the Cluster ID from the kube-system namespace yaml description in the Sos Report.
+    [Arguments]    ${sos_report_tarfile}
+
+    ${sos_report_untared}=    Extract Sos Report    ${sos_report_tarfile}
+
+    ${namespace_yaml_path}=    Set Variable
+    ...    ${sos_report_untared}/sos_commands/microshift/namespaces/${CLUSTERID_NS}/${CLUSTERID_NS}.yaml
+    ${clusterid}    ${rc}=    Execute Command
+    ...    cat ${namespace_yaml_path} | sed -n 's/\\s\\suid:\\s//p'
+    ...    sudo=True    return_rc=True    return_stdout=True
+    Should Be Equal As Integers    0    ${rc}
+
+    Should Not Be Empty    ${clusterid}
+    RETURN    ${clusterid}
+
 Tamper With Cluster ID File
     [Documentation]    Append invalid characters to the cluster ID file.
-    ${stdout}    ${stderr}    ${rc}=    Execute Command
+    ${rc}=    Execute Command
     ...    sed -i '$ s/$/123/' ${CLUSTERID_FILE}
-    ...    sudo=True    return_rc=True    return_stdout=True    return_stderr=True
+    ...    sudo=True    return_rc=True    return_stdout=False
+
+    Should Be Equal As Integers    0    ${rc}
+
+Remove Cluster ID File
+    [Documentation]    Append invalid characters to the cluster ID file.
+    ${rc}=    Execute Command
+    ...    rm -rf ${CLUSTERID_FILE}
+    ...    sudo=True    return_rc=True    return_stdout=False
 
     Should Be Equal As Integers    0    ${rc}
+
+Create Sos Report
+    [Documentation]    Create a MicroShift Sos Report and return the tar file path
+
+    ${rand_str}=    Generate Random String    4    [NUMBERS]
+    ${sos_report_dir}=    Catenate    SEPARATOR=    /tmp/rf-test/sos-report_    ${rand_str}
+
+    ${rc}=    Execute Command
+    ...    mkdir -p ${sos_report_dir}
+    ...    sudo=True    return_rc=True    return_stdout=False
+    Should Be Equal As Integers    0    ${rc}
+
+    ${rc}=    Execute Command
+    ...    sos report --batch --all-logs --tmp-dir ${sos_report_dir} -p microshift -o logs
+    ...    sudo=True    return_rc=True    return_stdout=False
+    Should Be Equal As Integers    0    ${rc}
+
+    ${sos_report_tarfile}    ${rc}=    Execute Command
+    ...    find ${sos_report_dir} -type f -name "sosreport-*.tar.xz"
+    ...    sudo=True    return_rc=True    return_stdout=True
+    Should Be Equal As Integers    0    ${rc}
```

Comment:
We can simplify this code by using `Command Should Work` from `microshift-host.resource`

Reply:
thanks, I agree, I didn't know about this keyword